Sources/NIOPosix/LinuxUring.swift

```diff
@@ -312,239 +335,314 @@ final internal class URing {
                 flags |= IORING_POLL_ADD_MULTI       // ask for multiple updates
             }
 
-            CNIOLinux.io_uring_prep_poll_update(sqe, .init(userData: bitpattern), .init(userData: bitpattern), newPollmask, flags)
-            CNIOLinux.io_uring_sqe_set_data(sqe, .init(userData: userbitPattern))
+            CNIOLinux.io_uring_prep_poll_update(sqe, oldUserData, newUserData, newPollMask, flags)
+            //CNIOLinux.io_uring_sqe_set_data64(sqe, newUserData) ??
         }
-        
-        if submitNow {
-            self.io_uring_flush()
+    }
+
+    func prep_write(fileDescriptor: CInt, pointer: UnsafeRawBufferPointer, registrationID: SelectorRegistrationID) {
+        let userData = UserData(fileDescriptor, registrationID, CQEEventType.write).asUInt64()
+
+        _debugPrint("URing.prep_write: " +
+            "fd[\(fileDescriptor)] " +
+            "pointer[\(pointer)] " +
+            "userData=[0x\(String(userData>>1, radix: 16))]")
+
+        assert(pointer.count > 0)
+
+        self.withSQE { sqe in
+            io_uring_prep_write(sqe, fileDescriptor, pointer.baseAddress, UInt32(pointer.count), 0)
+            io_uring_sqe_set_data64(sqe, userData)
         }
     }
 
-    internal func _debugPrint(_ s: @autoclosure () -> String)
-    {
-        #if SWIFTNIO_IO_URING_DEBUG_URING
-        print("L [\(NIOThread.current)] " + s())
-        #endif
+    func prep_writev(fileDescriptor: CInt, iovecs: UnsafeBufferPointer<IOVector>, registrationID: SelectorRegistrationID) {
+        let userData = UserData(fileDescriptor, registrationID, CQEEventType.write).asUInt64()
+
+        func bytesToWrite(_ iovecs: UnsafeBufferPointer<IOVector>) -> Int {
```

Comment:
It might be a little nicer to spell this in an `extension` on `UnsafeBufferPointer where Element == IOVector`.